src/sentry/services/hybrid_cloud/rpc.py

```diff
@@ -590,6 +591,19 @@ def _fire_request(self, headers: MutableMapping[str, str], data: bytes) -> reque
         except requests.exceptions.Timeout as e:
             raise self._remote_exception(f"Timeout of {settings.RPC_TIMEOUT} exceeded") from e
 
+    def _check_disabled(self):
+        if options.get("hybrid_cloud.rpc.disabled"):
+            raise RpcDisabledException("RPC disabled")
+
+        if disabled_service_methods := options.get("hybrid_cloud.rpc.disabled-service-methods"):
```

Comment:
This check isn't strictly needed but it'll save a lot of unnecessary work on the following line unless we're actively disabling service methods

Reply:
> If we entirely turn off RPC calls, most of the application becomes non-functional so I don't know if/when we would use that option.

The only real scenario I was thinking of was disabling a region to save control+other region.

Reply:
Do we want a safeguard against setting the "disabled" flag on the control silo? We could make it work only in region mode.